docs/src/modules/spring/pages/actions-publishing-subscribing.adoc

```diff
@@ -1 +1,117 @@
+
+
+== Subscribing to a Topic
+
+To receive messages from a Google Cloud Pub/Sub or Apache Kafka topic, annotate the service method `@Subscribe.Topic` and specify the topic name.
+
+The events from the topic are delivered to the Action and in this simplified example they are only logged.
+
+[source,java,indent=0]
+.src/main/java/com/example/actions/SubscribeAction.java
+----
+include::example$spring-eventsourced-counter/src/main/java/com/example/actions/SubscribeAction.java[tags=class;sub-topic-action]
+----
+<1> subscribing to the events from topic 'counter-events'. 
+<2> setting the method input type to the events produced by the counter.
+<3> any return is valid. See NOTE in Subscribing to an Event Sourced Entity.
+
+
+IMPORTANT: By default, Kalix assumes the messages in the topic were serialized as JSON and as such deserialize them into `CounterEvent`. You can find here more about xref:serialization.adoc[]
+
+== Receiving messages from an external Topic
+
+In the above example, you consumed JSON messages from a topic that you control. If you are consuming an external topic, the message format may not be under your control and may not be JSON. If this is the case the Spring SDK can't consume from that topic. 
+
+TODO https://github.com/lightbend/kalix/issues/7810
+
+== Type level annotations for subscribing 
+
+You can subscribe to a topic or an Event Source Entity by adding @Subscribe as a type level annotation, at the top of the class. This provides additional functionality for subscribing: all methods returning `Action.Effect` are selected to process incoming events. The Action will fail if it receives an event for which there is no method handler, unless the subscription is set with `ignoreUnknown = true`. 
+
+In the following example you can take a look at how the Action is configured to ignore unknown messages because it has only one method handler for `ValueIncrease`.    
+
+[source,java,indent=0]
+.src/main/java/com/example/actions/SubscribeTypeLevelAction.java
+----
+include::example$spring-eventsourced-counter/src/main/java/com/example/actions/SubscribeTypeLevelAction.java[tag=class]
+----
+<1> set to ignore unknown events
+<2> only processsing `ValueIncreased` events
+<3> any return is valid. See NOTE in Subscribing to an Event Sourced Entity.
+
+If you do not add `ignoreUnknown=true`, the action would fail when processing a `ValueMultiply`. By default is false.
+
+== Subscribing and acting upon
+
+Another possible usage for Actions is to consume events and act upon.
+
+For example, you may consume events from one entity or from a topic, transform to commands and send to an another entity or an external system. This is similar to the usage explained in xref:actions-as-controller.adoc[Actions as Controller], except that the Action is driven by the flow of incoming events instead of external user requests.
```

Comment:
Agree, although I think this is a general idea. It may be useful for the reader to abstract upon the first example. FYI This is how is it in the original docs. Would you like to remove it?